crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -800,6 +834,27 @@ macro_rules! timing_trace_out {
 }
 
 impl actor::HcP2p for HolochainP2pActor {
+    #[cfg(feature = "test_utils")]
+    fn test_set_full_arcs(&self, space: SpaceId) -> BoxFut<'_, ()> {
+        Box::pin(async {
+            for agent in self
+                .kitsune()
+                .unwrap()
+                .space(space)
+                .await
+                .unwrap()
+                .local_agent_store()
+                .get_all()
+                .await
+                .unwrap()
+            {
+                agent.set_cur_storage_arc(DhtArc::FULL);
+                agent.set_tgt_storage_arc_hint(DhtArc::FULL);
+                agent.invoke_cb();
```

Comment:
The peer_store `get_near_location` function was always returning no results, since agents now start with empty arcs... this was my solution to that for testing.

Reply:
In theory this all works itself out and just takes a bit of time.

- The first gossip round between a pair of peers should get them in sync, and then
- They will declare themselves as being at their target arc, and then
- They will publish their updated agent info

I'd prefer to start setting up test tooling to operate that way now. So we'd need to make sure that the two gossip configs are set to low values (10s or 100s of milliseconds) and then retry get operations in tests.

Reply:
If that's not working, I'd like to go and look at why in K2 in parallel

Reply:
There is no gossip here, it is using `CoreGossipStub` per the `default_test_builder()`.

Moreover, Gossip is a huge complicated emergent ball of problems. I don't think testing it belongs in a unit test for a mid-level crate like this. It seems to me we should have unit tests for gossip, then integration tests for gossip, but nothing in between.